// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

import "secretflow_serving/config/server_config.proto";
import "secretflow_serving/config/feature_config.proto";
import "secretflow_serving/config/model_config.proto";
import "secretflow_serving/config/cluster_config.proto";

package secretflow.serving.kuscia;

message KusciaServingConfig {
  message PartyConfig {
    ServerConfig server_config = 1;
    ModelConfig model_config = 2;
    FeatureSourceConfig feature_source_config = 3;
    ChannelDesc channel_desc = 4;
  }

  // Key: Party id
  // Value: Party config
  map<string, PartyConfig> party_configs = 1;

  // Indicate which parties can initiate the prediction
  // Default: All
  repeated string predictor_parties = 2;
}
